datafusion/core/src/physical_optimizer/projection_pushdown.rs

```diff
@@ -1345,7 +1385,9 @@ mod tests {
             Arc::new(NegativeExpr::new(Arc::new(Column::new("f", 4)))),
             Arc::new(ScalarFunctionExpr::new(
                 "scalar_expr",
-                Arc::new(|_: &[ColumnarValue]| unimplemented!("not implemented")),
+                ScalarFunctionDefinition::UDF(Arc::new(ScalarUDF::new_from_impl(
+                    DummyUDF::new(),
+                ))),
```

Comment:
Since here `ScalarFunctionExpr` needs a `ScalarFunctionDefinition`, I created one for each. Don't know whether there's a better way to do it.